### PR TITLE
feat: add new parameters to ``discord.Embed``

### PR DESCRIPTION
**Pull request by davidhozic**
## Summary
This PR adds parameters to ``discord.Embed`` as proposed in #1995.
It also adds 2 new classes to support the latter. The classes are: ``EmbedFooter`` and ``EmbedAuthor``

Example:
```py
import discord
import secret


embed = discord.Embed(
    author=discord.EmbedAuthor(
        name="Mr. President",
        icon_url="https://media.discordapp.net/attachments/1044657935608987808/1065995582654664724/svetlogo.png"
    ),
    footer=discord.EmbedFooter(text="This is a footer")
)


client = discord.Client()


@client.listen("on_ready")
async def start():
    channel = client.get_channel(1091835594243592293)
    await channel.send(embed=embed)
    await client.close()


client.run(secret.TOKEN)
```

which produces

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

File: discord/embeds.py [hunk 1]
```diff
@@ -103,6 +105,61 @@ class _EmbedAuthorProxy(Protocol):
...
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    name: :class:`str`
+        The name of the author.
+    url: :class:`str`
+        The url of the hyperlink created on the author name.
+    icon_url: :class:`str`
+        Url of the author icon image.
```
Comment (JustaSqu1d) [inline discord/embeds.py:120]:
```suggestion
        The URL of the author icon image.
```

File: discord/embeds.py [hunk 2]
```diff
@@ -103,6 +105,61 @@ class _EmbedAuthorProxy(Protocol):
...
+    """Represents the author on the :class:`Embed` object.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    name: :class:`str`
+        The name of the author.
+    url: :class:`str`
+        The url of the hyperlink created on the author name.
```
Comment (JustaSqu1d) [inline discord/embeds.py:118]:
```suggestion
        The URL of the hyperlink created in the author's name.
```

File: discord/embeds.py [hunk 3]
```diff
@@ -103,6 +105,61 @@ class _EmbedAuthorProxy(Protocol):
...
+    """Represents the footer on the :class:`Embed` object.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    text: :class:`str`
+       The text inside the footer.
+    icon_url: :class:`str`
+        Url of the footer icon image.
```
Comment (JustaSqu1d) [inline discord/embeds.py:147]:
```suggestion
        The URL of the footer icon image.
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
Also, note the change in their respective properties.

**Comment by JustaSqu1d** [comment]
Should we deprecate `set_author`, etc?

**Comment by davidhozic** [comment]
> Should we deprecate `set_author`, etc?

I think that's a good idea yes

**Comment by davidhozic** [comment]
> > Should we deprecate `set_author`, etc?
> 
> I think that's a good idea yes

On second thought, I'd leave it in, you still have the ``set_field_at`` and ``remove_field`` and other ``remove_`` methods available. 
Maybe you for example want to modify an embed from a received message and send it back, so these methods can be helpful.

**Comment by Lulalaby** [comment]
> Should we deprecate `set_author`, etc?

No, not in every case you want to use the constructor.
And what if you want to modify it afterwards

**Review by JustaSqu1d — APPROVED** [review]

**Review by Lulalaby — DISMISSED** [review]

**Comment by Nzii3** [comment]
I mean, I don't really see the point in this but, idk.

**Review by Nzii3 — APPROVED** [review]

**Review by plun1331 — COMMENTED** [review]
Ideally, embed.footer and embed.author should return their respective classes

**Comment by davidhozic** [comment]
Not sure what ``proxy_icon_url`` is supposed to be, but I'll leave it in the constructor since it's needed for the ``@properties`` and it will give you an error anyways if you try to provide it since ``Embed`` internally calls the ``set_`` methods in it's constructor.

**Comment by plun1331** [comment]
> Not sure what `proxy_icon_url` is supposed to be, but I'll leave it in the constructor since it's needed for the `@properties` and it will give you an error anyways if you try to provide it manually since `Embed` internally calls the `set_` methods in it's constructor.

it's a value returned by discord that proxies the image through their cdn iirc

**Comment by Lulalaby** [comment]
> it's a value returned by discord that proxies the image through their cdn iirc

That's correct

**Review by JustaSqu1d — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**